### PR TITLE
General fixes to DG

### PR DESCRIPTION
**Pull request by stanley-1**
- UCs have been shuffled and reorganized.
- New UCs have been added (for the pin, unpin, share, profile picture features)
- Visit this [link](https://stanley-1.github.io/tp/DeveloperGuide.html) to view the DG via GitHub pages hosted on my repo

### CODE REVIEW

**Comment by stanley-1** [comment]
- Temporarily marked as draft to facilitate the replacement of references to `AddressBook` in text with `SociaLite`. 
- Links to relevant code files will also be updated in this PR

**Comment by stanley-1** [comment]
Updated DG consists of the amendments as outlined in my previous comments attached to this PR. In addition, the `Architecture` section and a UML diagram have been updated. Lastly, outdated user stories have been deleted or re-organized.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -641,14 +590,14 @@ _Customization of Contacts_
...
 
 *MSS*
 
-1.  User requests to **list contacts (UC02)** or **find contact (UC04)**
+1.  User requests to **list contacts (UC02)** or **find contact (UC03)**
 2.  SociaLite returns a list of contacts according to the UC called
-3.  User specifies the index of his desired contact
-4.  SociaLite opens a contact card which presents all previously stored details associated with the contact
+3.  User specifies contact to be pinned at the top of the list
+4.  SociaLite pins the contact at the top of the list
```
Comment (david-eom) [inline docs/DeveloperGuide.md:600]: Small nit here: maybe we should use "pin to the top..." to standardise with the use case title!

Comment (david-eom) [reply docs/DeveloperGuide.md:600]: Actually let me do it myself in a separate PR! Since this section should fall under me haha

**Review by david-eom — APPROVED** [review]
Hi Stanley, thanks for the hard efforts to update the DG! The diagrams looks on point, only manage to find out one small nit but that should be under my own section, will be making another PR later!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (stanley-1) [reply docs/DeveloperGuide.md:600]: Roger boss!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -689,81 +642,77 @@ _Customization of Contacts_
...
-* 2a. No contacts match all given keywords, blank list is returned
+*MSS*
+
+1.  User types in command to exit application
+2.  SociaLite closes
 
     Use case ends.
 
+
 *{More to be added}*
```
Comment (Zacchua) [inline docs/DeveloperGuide.md:716]: Haha do we still need the more to be added since this is the last iteration?

**Review by Zacchua — APPROVED** [review]
Great job on this Stanley! Your attention to detail is really amazing! Just left a small comment below about the "More use cases to be added" statement

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (stanley-1) [reply docs/DeveloperGuide.md:724]: HAHA good point! maybe @david-eom can delete this line in his PR for the DG later?

**Comment by stanley-1** [comment]
Merge conflict has been resolved and PR is ready for merging into master if you guys are fine with it! @Zacchua @david-eom Thank you!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (david-eom) [reply docs/DeveloperGuide.md:724]: Alright! Will do for all the `{to be added}` stuff later in my PR!

Comment (stanley-1) [reply docs/DeveloperGuide.md:724]: Thanks!

**Pull request merged**